runtime/ms_rest/lib/ms_rest/serialization.rb

```diff
@@ -391,7 +391,14 @@ def serialize_sequence_type(mapper, object, object_name)
       # @param model_name [String] Name of the model to retrieve.
       #
       def get_model(model_name)
-        Object.const_get(@context.class.to_s.split('::')[0...-1].join('::') + "::Models::#{model_name}")
+        consts = @context.class.to_s.split('::')
+        if consts.any?{ |const| const == 'Models' }
```

Comment:
Uhm, sure. Basically, you'd change from:
```ruby
      # Serialize Request
      request_mapper = VirtualMachine.mapper()
      request_content = @client.serialize(request_mapper,  parameters, 'parameters')
      request_content = request_content != nil ? JSON.generate(request_content, quirks_mode: true) : nil
      # ....
      result_mapper = VirtualMachine.mapper()
      result.body = @client.deserialize(result_mapper, parsed_response, 'result.body')
```
To:
```ruby
      # Serialize Request
      request_content = VirtualMachine.serialize(parameters, name: 'parameters', quirks_mode: true)
      # ....
      result.body = VirtualMachine.deserialize(parsed_response, name: 'result.body')
```
The client really doesn't need to be involved. It's coupled to serialization only because the deserializer needs to know how to retrieve the model type. If the serialization logic is decorated on the model, this becomes trivial, insatiate a `self.class`.